### PR TITLE
Support early xz/lzma decompression, and use it wherever possible

### PR DESCRIPTION
**Pull request by Googulator**

### CODE REVIEW

**Review by stikonas — APPROVED** [review]

File: steps/make-3.82/sources [hunk 1]
```diff
@@ -1 +1 @@
-https://mirrors.kernel.org/gnu/make/make-3.82.tar.bz2 e2c1a73f179c40c71e2fe8abf8a8a0688b8499538512984da4a76958d0402966
+http://mirror.clarkson.edu/slitaz/sources/packages/m/make-3.82.tar.lzma 9475ef76dd33c319aa8a48d8bca28b2312b90b5e59c2b0b05b206c3f50614ee2
```
Comment (stikonas) [inline steps/make-3.82/sources:1]: oh, this one is a bit strange. Looks like the only source for lzma tarball? Did they repackage it themselves?
@fosslinux what do you think about this one?

Comment (Googulator) [reply steps/make-3.82/sources:1]: Right, I wasn't able to find any other source for this tarball.

Comment (fosslinux) [reply steps/make-3.82/sources:1]: Hm, I think I'd prefer to use the more widespread bz2, not sure it's worth saving 252KB, but happy to hear other options.

**Review by fosslinux — APPROVED** [review]
I'm cool with this :)

**Pull request merged**